### PR TITLE
Document `scipy.fft` backend usage

### PR DESCRIPTION
**Pull request by leofang**
Related to https://github.com/scipy/scipy/issues/14266#issuecomment-880029174. We should present at least a self-contained snippet to make it more clear.

cc: @AnirudhDagar @peterbell10 @rgommers

### CODE REVIEW

**Comment by leofang** [comment]
Hmmm the doctest disappeared...Let me try power-cycling.

**Comment by leofang** [comment]
The change is rendered here: https://cupy--5514.org.readthedocs.build/en/5514/user_guide/fft.html

File: docs/source/user_guide/fft.rst [hunk 1]
```diff
@@ -16,6 +16,49 @@ or may be absent in `hipFFT`_/`rocFFT`_ targeting AMD GPUs.
...
+    import cupyx.scipy.fft as cufft
+    import scipy.fft
+    scipy.fft.set_global_backend(cufft)
+
+    a = cp.random.random(100).astype(cp.complex64)
+    b = scipy.fft.fft(a)  # equivalent to cufft.fft(a)
+
+.. note::
+
+    Currently in case of conflicts the precedence of the three APIs is :func:`~scipy.fft.set_backend` > :func:`~scipy.fft.set_global_backend`
```
Comment (AnirudhDagar) [inline docs/source/user_guide/fft.rst:52]: This order is true in general but might not be true if the user decides to use `try_last=True`. Should we mention that somewhere? @peterbell10 @leofang

**Review by AnirudhDagar — COMMENTED** [review]
Overall looks good to me, thanks @leofang
I just made a small comment.

File: docs/source/user_guide/fft.rst (hunk 1 above)
Comment (leofang) [reply docs/source/user_guide/fft.rst:52]: Thanks, @AnirudhDagar! Given that in the next sentence I referred to the SciPy documentation, I think this sentence should just be removed. It's SciPy's responsibility to explain implementation details such as the precedence, not CuPy's. If it'd be ever changed one day our docs would be outdated.

Comment (AnirudhDagar) [reply docs/source/user_guide/fft.rst:52]: Yeah, this sounds like a better idea.

**Review by toslunar — APPROVED** [review]
Thank you everyone! LGTM

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
@toslunar btw this PR can be backported.

**Comment by leofang** [comment]
Thanks, @AnirudhDagar @toslunar!

**Pull request merged**